gateway/src/main/java/io/camunda/zeebe/gateway/impl/configuration/CompressionCfg.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.gateway.impl.configuration;
+
+import io.atomix.cluster.messaging.MessagingConfig.CompressionAlgorithm;
+import java.util.Objects;
+
+public class CompressionCfg {
+  private static final boolean DEFAULT_ENABLE_COMPRESSION = false;
+
+  private boolean enableCompression = DEFAULT_ENABLE_COMPRESSION;
+  private CompressionAlgorithm algorithm = CompressionAlgorithm.GZIP;
+
+  public boolean isEnableCompression() {
+    return enableCompression;
+  }
+
+  public void setEnableCompression(final boolean enableCompression) {
+    this.enableCompression = enableCompression;
+  }
+
+  public CompressionAlgorithm getAlgorithm() {
+    return algorithm;
+  }
+
+  public void setAlgorithm(final CompressionAlgorithm algorithm) {
+    this.algorithm = algorithm;
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(enableCompression, algorithm);
```

Comment:
:upside_down_face: strictly speaking, this isn't the IntelliJ default style: https://github.com/camunda-cloud/zeebe/wiki/Code-Style#equalshashcode-implementations

Same goes for `MembershipCfg`

Reply:
:+1: Other configuration classes uses this format though. I will update the new code to follow the code style.